src/App.jsx

```diff
@@ -14,33 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortOrder, setSortOrder] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (isReversed) sortedGoods.reverse();
+    setGoods(sortedGoods);
+    setSortOrder('alphabet');
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (a, b) => a.length - b.length,
+    );
+
+    if (isReversed) sortedGoods.reverse();
+    setGoods(sortedGoods);
+    setSortOrder('length');
+  };
+
+  const reverseGoods = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setSortOrder(null);
+    setIsReversed(false);
+  };
+
+  const isResetVisible =
+    JSON.stringify(goods) !== JSON.stringify(goodsFromServer);
```

Comment:
Good execution, but it's better to put all the sorting algorithms as well as data preparation in to the outer function. 

Also try using switch case for alphabetical and length sorting

It should look something like that:
```jsx
function preparedGoodsData(goods, sortName, isReversed) {
  const preparedGoods = [...goods];

  if (sortValue) {
    preparedGoods.sort((good1, good2) => {
      switch (sortName) {
        case SORT_BY_NAME:
          ...;

        case SORT_BY_LENGTH:
          ...;

        default:
          return 0;
      }
    });
  }

  if (isReversed) {
    ...;
  }

  return preparedGoods;
}


export const App = () => {
  ...;
  const visibleGoods = preparedGoodsData(goodsFromServer, sortName, isReversed);
  ...;

Reply:
good point